### PR TITLE
feat(Icons): direct import on icons instead of icon component

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Refactor

#### What is expected?

New big change in icons, we added a new way to import icons. You can now directly import the icon name through `@ultraviolet/icons`.

```tsx
// Before
import { Icon } from '@ultraviolet/icons';

const MyComponent = () => (
  <Icon name="address" sentiment="danger" size="small" />
);
```

```tsx
// Now
import { Address } from '@ultraviolet/icons';
// OR
import { Address } from '@ultraviolet/icons/Address';

const MyComponent = () => (
  <Address sentiment="danger" size="small" />
);
```

This change will make it easier to use icons in your project and reduce the bundle size.
Exact same pattern is changed for `<CategoryIcon />` and `<ProductIcon />`:

```tsx
// Before
import { CategoryIcon } from '@ultraviolet/icons';

// Now
import { Baremetal } from '@ultraviolet/icons/category';
// OR
import { Baremetal } from '@ultraviolet/icons/category/Baremetal';
```

```tsx
// Before
import { ProductIcon } from '@ultraviolet/icons';

// Now
import { Instance } from '@ultraviolet/icons/product';
// OR
import { Instance } from '@ultraviolet/icons/product/Instance';
```

### CODE REVIEW

**Comment by atissedrescaleway** [comment]
As discussed with @philibea in private : we'll need a solution to be able to import an icon dynamically, without including all the icons in the final bundle. 

from our side we'll be able to use [this Next.js feature](https://vercel.com/blog/how-we-optimized-package-imports-in-next-js#new-solution-optimizepackageimports) so we can purge the unused icons from the final bundle.

That requires being able to import all the icons from a barrel file, like with material icons: 

import * as MuiIcons from '@material-ui/icons'

```
export default ({ iconName }) => (
  <div>
    // component content
    {MuiIcons[iconName]}
  </div>
)
```

File: scripts/generate-icons-file.mjs [hunk 1]
```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-await-in-loop */
+import { promises as fs } from 'fs'
```
Comment (philibea) [inline scripts/generate-icons-file.mjs:2]:
```suggestion
import { promises as fs } from 'node:fs'
```

File: scripts/generate-icons-file.mjs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import path from 'path'
```
Comment (philibea) [inline scripts/generate-icons-file.mjs:3]:
```suggestion
import path from 'node:path'
```

File: scripts/generate-icons-file.mjs [hunk 3]
```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-await-in-loop */
```
Comment (philibea) [inline scripts/generate-icons-file.mjs:1]: can we do a ts file

Comment (philibea) [inline scripts/generate-icons-file.mjs:1]: you can deactivate this rules for all ./scrtips inside eslint config

File: packages/icons/package.json [hunk 4]
```diff
@@ -45,6 +45,36 @@
...
       "require": "./dist/index.cjs",
       "default": "./dist/index.js"
+    },
+    "./*": {
+      "types": "./dist/components/Icon/__generatedIcons__/*.d.ts",
+      "require": "./dist/components/Icon/__generatedIcons__/*.cjs",
+      "default": "./dist/components/Icon/__generatedIcons__/*.js"
+    },
+    "./product": {
+      "types": "./dist/components/ProductIcon/index.d.ts",
```
Comment (johnrazeur) [inline packages/icons/package.json:55]: We should `import`

File: packages/icons/package.json [hunk 5]
```diff
@@ -44,7 +44,44 @@
     ".": {
       "types": "./dist/index.d.ts",
       "require": "./dist/index.cjs",
-      "default": "./dist/index.js"
+      "default": "./dist/index.js",
```
Comment (philibea) [inline packages/icons/package.json:47]: default should be the last

File: packages/icons/package.json [hunk 6]
```diff
@@ -44,7 +44,44 @@
...
+      "import": "./dist/index.js"
+    },
+    "./*": {
+      "types": "./dist/components/Icon/__generatedIcons__/*.d.ts",
+      "require": "./dist/components/Icon/__generatedIcons__/*.cjs",
+      "default": "./dist/components/Icon/__generatedIcons__/*.js",
+      "import": "./dist/components/Icon/__generatedIcons__/*.js"
```
Comment (philibea) [inline packages/icons/package.json:54]:
```suggestion
      "import": "./dist/components/Icon/__generatedIcons__/*.js",
       "default": "./dist/components/Icon/__generatedIcons__/*.js"
```

File: package.json [hunk 7]
```diff
@@ -166,6 +167,7 @@
     "size-limit": "11.1.4",
     "storybook": "8.2.9",
     "timekeeper": "2.3.1",
+    "tsx": "^4.19.0",
```
Comment (philibea) [inline package.json:170]:
```suggestion
    "tsx": "4.19.0",
```

File: packages/icons/package.json (hunk 5 above)
Comment (philibea) [reply packages/icons/package.json:47]: https://publint.dev/

**Review by philibea — APPROVED** [review]

**Review by johnrazeur — APPROVED** [review]

**Pull request merged**